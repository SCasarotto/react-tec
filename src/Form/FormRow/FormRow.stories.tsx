import React from 'react';

import { Story, Meta } from '@storybook/react';

import { Input } from '../Input';
import { FormRow, FormRowProps } from './FormRow';

export default {
  title: 'Form/Row',
  component: FormRow,
  argTypes: {},
} as Meta;

const Template: Story<FormRowProps> = (args) => <FormRow {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: <Input onChange={(e) => console.log(e.target.value)} />,
};

export const Sizes = () => (
  <>
    <FormRow>
      <Input onChange={(e) => console.log(e.target.value)} />
    </FormRow>
    <FormRow rowSize='half'>
      <Input onChange={(e) => console.log(e.target.value)} />
    </FormRow>
    <FormRow rowSize='half' last>
      <Input onChange={(e) => console.log(e.target.value)} />
    </FormRow>
    <FormRow rowSize='two-third'>
      <Input onChange={(e) => console.log(e.target.value)} />
    </FormRow>
    <FormRow rowSize='third' last>
      <Input onChange={(e) => console.log(e.target.value)} />
    </FormRow>
    <FormRow rowSize='half'>
      <Input onChange={(e) => console.log(e.target.value)} />
    </FormRow>
    <FormRow rowSize='forth'>
      <Input onChange={(e) => console.log(e.target.value)} />
    </FormRow>
    <FormRow rowSize='forth' last>
      <Input onChange={(e) => console.log(e.target.value)} />
    </FormRow>
    <FormRow rowSize='condensed'>
      <Input onChange={(e) => console.log(e.target.value)} />
    </FormRow>
    <FormRow rowSize='condensed'>
      <Input onChange={(e) => console.log(e.target.value)} />
    </FormRow>
    <FormRow rowSize='condensed' last>
      <Input onChange={(e) => console.log(e.target.value)} />
    </FormRow>
  </>
);
