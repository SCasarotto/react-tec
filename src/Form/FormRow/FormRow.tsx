import React from 'react';

import { StyledRow } from './styledComponents';

export type FormRowSize =
  | 'full'
  | 'two-third'
  | 'half'
  | 'third'
  | 'forth'
  | 'condensed';
export interface FormRowCustomProps {
  rowSize?: FormRowSize;
  last?: boolean;
}
export interface FormRowProps
  extends React.HTMLAttributes<HTMLDivElement>,
    FormRowCustomProps {}
/**
 *
 * Form row component. `Row` provides a simple way to layout form rows. Easily enabling multiple inputs or buttons on the same row.
 *
 */
export const FormRow = React.forwardRef<HTMLDivElement, FormRowProps>(
  (props, ref) => {
    const { rowSize = 'full', className = '', ...rest } = props;
    return (
      <StyledRow
        className={`tec-formRow ${className}`}
        rowSize={rowSize}
        ref={ref}
        {...rest}
      />
    );
  },
);
