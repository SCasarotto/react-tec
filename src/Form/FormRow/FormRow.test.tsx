import React from 'react';

import { render, waitFor } from '@testing-library/react';

import { FormRow } from './FormRow';

describe('FormRow', () => {
  // Renders Without Error
  it('matches snapshot', () => {
    const { asFragment } = render(<FormRow />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Class Name
  it('accepts custom className', () => {
    const { container } = render(<FormRow className='myClassName' />);
    expect(container.firstChild).toHaveClass('myClassName');
  });

  // Ref
  it('renders with ref', async () => {
    const ref: { current: HTMLDivElement | null } = { current: null };
    const { getByTestId } = render(
      <FormRow ref={ref} data-testid='myFormRow' />,
    );
    getByTestId('myFormRow');
    // Using this as the focus check didn't seem to work
    waitFor(() => expect(ref.current).toBeTruthy());
  });
});
