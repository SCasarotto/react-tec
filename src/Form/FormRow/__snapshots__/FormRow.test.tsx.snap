// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormRow matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  margin-bottom: 10px;
  text-align: left;
  box-sizing: border-box;
  padding-right: 0px;
  width: 100%;
}

<div
    class="c0 tec-formRow "
  />
</DocumentFragment>
`;
